osu.Game.Rulesets.Osu/Edit/Blueprints/Sliders/Components/PathControlPointPiece.cs

```diff
@@ -195,13 +197,36 @@ private void updateMarkerDisplay()
 
             markerRing.Alpha = IsSelected.Value ? 1 : 0;
 
-            Color4 colour = ControlPoint.Type.Value != null ? colours.Red : colours.Yellow;
+            Color4 colour = getColourFromNodeType();
 
             if (IsHovered || IsSelected.Value)
                 colour = colour.Lighten(1);
 
             marker.Colour = colour;
             marker.Scale = new Vector2(slider.Scale);
         }
+
+        private Color4 getColourFromNodeType()
+        {
+            if (!(ControlPoint.Type.Value is PathType pathType))
+                return colours.Yellow;
+
+            switch (pathType)
+            {
+                case PathType.Catmull:
+                    return colours.Seafoam;
+
+                case PathType.Bezier:
+                    return colours.Pink;
+
+                case PathType.PerfectCurve:
+                    return colours.PurpleDark;
+
+                default:
+                    return colours.Red;
+            }
+        }
+
+        public string TooltipText => ControlPoint.Type.Value.ToString() ?? string.Empty;
```

Comment:
Perfect curve points get displayed as `PerfectCurve`, instead of a humanized form `Perfect curve`.
```suggestion
        public string TooltipText => ControlPoint.Type.Value.ToString()?.Humanize() ?? string.Empty;
```

Reply:
Was aware of this, but don't think it's that important. If anything it should be just `Perfect` or even something that is more understandable to the user like `Circular`.